pkg/graph/resolvers/mto_milestone_test.go

```diff
@@ -0,0 +1,165 @@
+	categories, err = MTOCategoryGetByModelPlanIDLOADER(suite.testConfigs.Context, plan.ID)
+	suite.NoError(err)
+	suite.Len(categories, 2) // 1 new category (Operations) + Uncategorized
+
+	operationsCategory, found := lo.Find(categories, func(item *models.MTOCategory) bool {
+		return item != nil && item.Name == "Operations"
+	})
+	suite.True(found)
+
+	operationsSubCategories, err := MTOSubcategoryGetByParentIDLoader(suite.testConfigs.Context, plan.ID, operationsCategory.ID)
+	suite.NoError(err)
+	suite.Len(operationsSubCategories, 2) // should have 'Participant and beneficiary tracking' + Uncategorized
+	participantSubcategory, found := lo.Find(operationsSubCategories, func(item *models.MTOSubcategory) bool {
+		return item != nil && item.Name == "Participant and beneficiary tracking"
+	})
+	suite.True(found)
+
+	if suite.NotNil(milestone.MTOCategoryID) {
+		suite.Equal(participantSubcategory.ID, *milestone.MTOCategoryID) // ensure that the common milestone was created pointed to the subcategory
+	}
+
+	// Finally, do the same, but ensure that if we would try to create a duplicate category by name, we fail silently
+	// For this example, models.MTOCommonMilestoneKeyAppSupportCon will attempt to create ('Operations','Internal functions'), but since
+	// the 'Operations' category already exists, we should only see a single new subcategory ('Internal functions')
+	milestone2, err := MTOMilestoneCreateCommon(suite.testConfigs.Context, suite.testConfigs.Logger, suite.testConfigs.Principal, suite.testConfigs.Store, plan.ID, models.MTOCommonMilestoneKeyAppSupportCon)
+	suite.NoError(err)
+	suite.Equal(suite.testConfigs.Principal.UserAccount.ID, milestone2.CreatedBy)
+	suite.Nil(milestone2.ModifiedBy)
+	suite.Nil(milestone2.ModifiedDts)
+
+	// Shouldn't have any new top-level categories, just 'Operations' and 'Uncategorized'
+	categories, err = MTOCategoryGetByModelPlanIDLOADER(suite.testConfigs.Context, plan.ID)
+	suite.NoError(err)
+	suite.Len(categories, 2) // 1 existing category (Operations) + Uncategorized
+
+	// Re-fetch subcategories to see if we see 3 total (since we added 'Internal functions')
+	operationsSubCategories, err = MTOSubcategoryGetByParentIDLoader(suite.testConfigs.Context, plan.ID, operationsCategory.ID)
+	suite.NoError(err)
+	suite.Len(operationsSubCategories, 3) // should have 'Participant and beneficiary tracking', 'Internal functions', + Uncategorized
+
+	// Find 'Participant and beneficiary tracking'
+	_, foundParticipant := lo.Find(operationsSubCategories, func(item *models.MTOSubcategory) bool {
+		return item != nil && item.Name == "Participant and beneficiary tracking"
+	})
+	suite.True(foundParticipant)
+
+	// Find 'Internal functions'
+	_, foundInternalFunctions := lo.Find(operationsSubCategories, func(item *models.MTOSubcategory) bool {
+		return item != nil && item.Name == "Internal functions"
+	})
+	suite.True(foundInternalFunctions)
```

Comment:
Would it be useful to validate a case where the  subcategory already exists? Or that it will make a new category if you rename a category?

Reply:
I might add some `TODO` comments to consider those cases! I didn't do those right away since I wanted to just kickstart these tests with the happy paths, but I agree these would be good test cases!

Reply:
e5a276b9b